File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.js
/**
 * @param {number[]} nums
 * @param {number} x
 * @return {number}
 */
var minOperations = function(nums, x) {
  const n = nums.length;
  let sum = nums.reduce((a, b) => a + b) - x;
  let [j, window] = [0, -1];
  
  if (sum < 0) return -1;
  if (!sum) return n;
  
  for (let i = 0; i < n; i++) {
    sum -= nums[i];
    while (sum < 0 && j < i) sum += nums[j++];
    
    if (!sum) window = Math.max(window, i - j + 1);
  }
  
  return window < 0 ? window : n - window;
};

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        s = sum(nums) - x
        n = len(nums)
        j = window = 0
        
        if s < 0:
            return -1
        if s == 0:
            return n
        
        for i, num in enumerate(nums):
            s -= num
            while s < 0:
                s += nums[j]
                j += 1
            if s == 0:
                window = max(window, i - j + 1)
        return -1 if window == 0 else n - window